packages/spacecat-shared-rum-api-client/src/functions/formVitals.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { FlatBundle } from '../common/flat-bundle.js';
+
+function collectFormVitals(groupedByUrlId) {
+  const { url, items: itemsByUrlId } = groupedByUrlId;
+
+  const result = itemsByUrlId
+    .flatMap((item) => item.items)
+    .reduce((acc, cur) => {
+      // Check for 'viewblock' checkpoint
+      if (cur && cur.checkpoint === 'viewblock' && cur.source === '.form') {
```

Comment:
Please include checkpoints like .marketo-form, .marketo, etc. and make configurable so we can continue to add more checkpoints.

Its also better the check if source contains form. @vdua WDYS?

Reply:
In our analysis we have figured out that even detecting `form` is not complete. eg: pfizer(litfulo) didn't have source having form in it.
We are going to detect it by using form element clicks - this is the most promising approach that has emerged from multiple analysis that I've done.

Reply:
@jalagari it would be a source of a checkpoint. Correct? Changed accordingly.
https://cq-dev.slack.com/archives/C07NX285LF3/p1730986280139059?thread_ts=1730983183.598219&cid=C07NX285LF3

Reply:
done